packages/cms-cli/lib/commonOpts.js

```diff
@@ -3,13 +3,21 @@ const {
   getPortalId: getPortalIdFromConfig,
   getPortalConfig,
   getAndLoadConfigIfNeeded,
+  loadConfigFromEnvironment,
   DEFAULT_MODE,
   Mode,
 } = require('@hubspot/cms-lib');
 const { LOG_LEVEL } = Logger;
 
 const addPortalOptions = program => {
   program.option('--portal <portal>', 'HubSpot portal id or name from config');
+  program.on('option:portal', () => {
+    if (loadConfigFromEnvironment()) {
+      throw new Error(
+        'Cannot specify a portal when environment variables are supplied. Please unset the environment variables or do not use the "--portal" flag.'
+      );
+    }
+  });
```

Comment:
Instead of leveraging the event, what if we adjusted the portal option to have custom processing or updated `validatePortal`.